fetcher/network.go

```diff
@@ -134,52 +158,69 @@ func (f *Fetcher) NetworkListRetry(
 			ctx,
 			metadata,
 		)
-		if errors.Is(err, ErrAssertionFailed) {
-			return nil, fmt.Errorf("%w: /network/list not attempting retry", err)
-		}
-
 		if err == nil {
 			return networkList, nil
 		}
 
+		if errors.Is(err.Err, ErrAssertionFailed) {
```

Comment:
nit: maybe we can apply your `checkError` method here

Reply:
hmm, the reason I didn't do it there is because we because we already return the response from our query if `err == nil` (line 162), and `checkError` just safely calls `errors.Is` anyway.